client/modules/workspace/src/SelectModal/SelectModal.tsx

```diff
@@ -109,83 +98,141 @@ const getScheme = (storageSystem: TTapisSystem): string => {
   return storageSystem.notes?.isMyData ? 'private' : 'public';
 };
 
-const getPath = (
-  storageSystem: TTapisSystem,
+const getSystemRootPath = (
+  storageSystem: TTapisSystem | undefined,
   user: TUser | undefined
 ): string => {
-  return storageSystem.notes?.isMyData
+  return storageSystem?.notes?.isMyData
     ? encodeURIComponent('/' + user?.username)
     : '';
 };
 
+const getBackPath = (
+  encodedPath: string,
+  searchTerm: string | null,
+  clearSearchTerm: () => void,
+  system: TTapisSystem | undefined,
+  user: TUser | undefined
+): string => {
+  if (searchTerm) {
+    clearSearchTerm();
+    return encodedPath;
+  }
+  const pathParts = decodeURIComponent(encodedPath).split('/');
+  const isRootPath = pathParts.join('/') === getSystemRootPath(system, user);
+  if (!isRootPath) {
+    pathParts.pop();
+  }
+  return encodeURIComponent(pathParts.join('/'));
+};
+
+const getParentFolder = (
+  name: string,
+  system: string,
+  path: string
+): TFileListing => {
+  return {
+    format: 'folder',
+    lastModified: '',
+    length: 1,
+    type: 'dir',
+    mimeType: '',
+    name: name,
+    permissions: '',
+    path: decodeURIComponent(path),
+    system: system,
+  };
+};
+
 function getFilesColumns(
   api: string,
-  system: string,
   path: string,
-  systemLabel: string,
   selectionMode: string,
+  searchTerm: string | null,
+  clearSearchTerm: () => void,
   selectionCallback: (path: string) => void,
-  navCallback: (path: string) => void
+  navCallback: (path: string) => void,
+  user: TUser | undefined,
+  selectedSystem?: TTapisSystem
 ): TFileListingColumns {
   return [
     {
       title: (
-        <HeaderTitle
-          api={api}
-          system={system}
-          path={path}
-          label={systemLabel}
-        />
-      ),
-      dataIndex: 'name',
-      ellipsis: true,
-
-      render: (data, record) =>
-        record.format === 'folder' ? (
+        <div>
           <Button
-            style={{
-              marginLeft: '3rem',
-              textAlign: 'center',
-              display: 'flex',
-              alignItems: 'center',
-            }}
-            onClick={() => navCallback(encodeURIComponent(record.path))}
             type="link"
+            onClick={() =>
+              navCallback(
+                getBackPath(
+                  path,
+                  searchTerm,
+                  clearSearchTerm,
+                  selectedSystem,
+                  user
+                )
+              )
+            }
           >
-            <i
-              role="none"
-              className="fa fa-folder-o"
-              style={{ color: '#333333', marginRight: '8px' }}
-            ></i>
-            {data}
+            <LeftOutlined />
+            Back
```

Comment:
Should "Back" be disabled if you are currently in root?

Reply:
Fixed.

Reply:
![Screenshot 2024-07-11 at 8 18 32 AM](https://github.com/DesignSafe-CI/portal/assets/2568355/638abc37-1786-400c-a659-e93ce87da722)
![Screenshot 2024-07-11 at 8 18 26 AM](https://github.com/DesignSafe-CI/portal/assets/2568355/58f3725d-2d1f-43af-b89d-a11ff18dffb7)